src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type.equals(Excavator.class)) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        } else if (type.equals(Truck.class)) {
+            return new ArrayList<>(new TruckProducer().get());
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
this can be simplified

you can init the returning list as empty and then just reassign for a corresponding producer

if no producer found - just return the original empty list

this way you'll have just 1 `return`, 1 `if`, 1 `else if` and `else` statements

and there is no sense to wrap `producer.get()` into additonal `ArrayList` - producers already return `List`